docs/6-developer-guide/2-testing.md

```diff
@@ -1,56 +1,61 @@
 # Code Testing
 
-:::caution Hard Hat Area
-This page is still being developed. More content will be added soon!
-:::
-
-
 Currently, we test Zarf through a series of end-to-end tests which can be found in the [e2e directory](https://github.com/defenseunicorns/zarf/tree/master/src/test/e2e) of the project. This directory holds all of our e2e tests that we use to verify Zarf functionality in an environment that replicates a live setting. The tests in this directory are automatically run against several K8s distros whenever a PR is opened or updated.
 
 ## Running Tests Locally
+
 The tests in this directory are also able to be run locally!
 
 ### Dependencies
-Running the tests locally have the same prerequisites as running and building Zarf:
- 1. GoLang >= `1.18.x`
- 2. Make
- 3. (for the `kind` and `k3d` options only) Docker
- 4. (for the `k3s` cluster only) Linux with root privileges
-
-### Existing K8s Cluster
-If you have a cluster already running, and use the env var `TESTDISTRO=provided`, the test suite will use the `KUBECONFIG` env var and the cluster that is currently configured as the active context. To make sure you are running in the right cluster, run something like `kubectl get nodes` with no optional flags and if the nodes that appear are the ones you expect, then the tests will use that cluster as well.
 
-This means that you are able to run the tests against any remote distro you want, like EKS, AKS, GKE, RKE, etc.
-
-### No Existing K8s Cluster
-If you do not have a local cluster running, no worries! The e2e tests use the `sigs.k8s.io/kind` and `github.com/k3d-io/k3d/v5` libraries to stand up local clusters to test against. All you have to do is make sure Docker is running and set the `TESTDISTRO` env var to either `"kind"` or `"k3d"` and the test suite will automatically create the appropriate cluster before the test run, run the tests on it, then automatically destroy it to clean up.
+Running the tests locally have the same prerequisites as running and building Zarf:
```

Comment:
Is `gcc` a dependency?

I pulled zarf over to a remote ubuntu dev machine (vanilla ec2) and installed GoLang/Make and ran into this error when running `APPLIANCE_MODE=false make test-e2e ARCH=amd64`

```
# runtime/cgo
cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in $PATH
FAIL    github.com/defenseunicorns/zarf/src/test/e2e [build failed]
make: *** [Makefile:108: test-e2e] Error 2
```

Reply:
`gcc` should not be a dependency as far as I'm aware. The only time `cgo` shows up in our repo is a comment saying we're doing something to avoid issues with it haha.

Did this failure happen right as you started the test before it actually did anything? My first suspicion is one of the `os.exec` commands we're running is using `gcc`/`cgo` under the hood

Reply:
Thanks for the affirmation.

I believe it happens during the `build-examples` section. Oddly I received it after the `podinfo-via-flux` component package build...

Reply:
Oh eww, seems to be a weirdness with Go..

When we build our binary we set the ENV `CGO_ENABLED=0`. Looks like when testing our code we will either need to have gcc installed or export that ENV.

https://github.com/golang/go/issues/28065#issuecomment-466523353


@brandtkeller can you try re-running with `APPLIANCE_MODE=false CGO_ENABLED=0 make test-e2e ARCH=amd64`

Reply:
That fixed it - all tests pass without gcc installed

Reply:
CGO_ENABLED is a *nix thing if I recall--would there be any known issue with adding this to our make statements for tests?  I'd be happy to update this or @brandtkeller if you have a suggestion for how to improve these docs further before we merge would be happy to do that to.